### ISSUE TITLE
relation returns `nil` for differently zoned zdts

### ISSUE THREAD

**Issue by SevereOverfl0w** [juxt/tick#133]
```
(t/relation
  (clojure.edn/read-string
    {:readers time-literals.read-write/tags}
    "#:tick{:beginning #time/zoned-date-time \"2021-02-24T00:00Z[GMT]\", :end #time/zoned-date-time \"2021-02-25T00:00Z[GMT]\"}")  
  (clojure.edn/read-string
    {:readers time-literals.read-write/tags}
    "#:tick{:beginning #time/zoned-date-time \"2021-02-23T00:00Z[Europe/London]\", :end #time/zoned-date-time \"2021-02-24T00:00Z[Europe/London]\"}"))
```

returns `nil`, should return `:met-by`

**Comment by jgoday** [comment]
Seems that the problem lies in using object equality for the "Allen's Basic Relations" predicates inside src/tick/interval.cljc.

I have made PR defining a new operator === in ITimeComparision protocol, and implementing it with ZonedDateTime iEqual method.

Is this a valid solution ?

**Comment by henryw374** [comment]
fixed by #9c76dcf
